lib/gollum/app.rb

```diff
@@ -179,7 +179,7 @@ def wiki_new
       halt 500 unless tempfile.is_a? Tempfile
 
       # Remove page file dir prefix from upload path if necessary -- committer handles this itself
-      dir      = wiki.per_page_uploads ? params[:upload_dest].match(/^(#{wiki.page_file_dir}\/+)?(.*)/)[2] : 'uploads'
+      dir      = wiki.per_page_uploads ? params[:upload_dest] : ::File.join([wiki.page_file_dir, 'uploads'].compact)
```

Comment:
On 05-10-15 12:21, Jiří Smolík wrote:
> Personally, I think it's really time to revise all path-related 
> features and come up with a unified, documented and enforced model...

Handling `page-file-dir` only internally will actually make this much 
easier, since we can then be sure that any request paths from the client 
don't contain the `page-file-dir` prefix yet. So there should just be a 
helper method on the `Wiki` class that converts request paths to in-repo 
paths (and one to do the opposite I guess). This being ruby I think 
there's no way to 'enforce' the use of a particular method to construct 
a path, nor should there be. But yeah, we can document it and make it 
explicit that developers should use the helper method.

For now we have a serious bug, and we should just squash it instead of 
waiting for someone to do the Right Thing Soon(tm).